File: lib/models/oplace.dart

class Oplace {
  String id;
  String name;
  String oTag;
  String description;
  List<String> products;

  Oplace({
    required this.id,
    required this.name,
    required this.oTag,
    required this.description,
    required this.products,
  });

  factory Oplace.fromJson(Map<String, dynamic> json) {
    return Oplace(
      id: json['id'],
      name: json['name'],
      oTag: json['o_tag'],
      description: json['description'],
      products: List<String>.from(json['products']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'o_tag': oTag,
      'description': description,
      'products': products,
    };
  }
}

File: backend/src/main/java/com/ordrweb/oplaces/Oplace.java
package com.ordrweb.oplaces;

import java.util.List;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

@Document
public class Oplace {
   
   public String name;
   @Indexed(unique = true) public String otag;
   public String description;
   public List<String> products;

   public Oplace() {
   }

   public Oplace(String name, String otag, String description, List<String> products) {
      this.name = name;
      this.otag = otag;
      this.description = description;
      this.products = products;
   }

   public String getName() {
      return this.name;
   }

   public String getO_tag() {
      return this.otag;
   }

   public String getDescription() {
      return this.description;
   }

   public List<String> getProdjucts() {
      return this.products;
   }

   public void setName(String name) {
      this.name = name;
   }

   public void setO_tag(String otag) {
      this.otag = otag;
   }

   public void setDescription(String description) {
      this.description = description;
   }

   public void setProduct(String product) {
      this.products.add(product);
   }

   public List<String> getProducts() {
      return this.products;
   }

   public void setProducts(final List<String> products) {
      this.products = products;
   }
}
